fenix/app/src/main/java/org/mozilla/fenix/settings/quicksettings/protections/ProtectionsView.kt

```diff
@@ -83,33 +85,51 @@ class ProtectionsView(
     private val shouldShowCookieBanner: Boolean
         get() = settings.shouldShowCookieBannerUI && settings.shouldUseCookieBanner
 
-    private fun bindCookieBannerProtection(isCookieBannerHandlingEnabled: Boolean) {
+    private fun bindCookieBannerProtection(cookieBannerMode: CookieBannerMode) {
         val context = binding.cookieBannerItem.context
         val label = context.getString(R.string.preferences_cookie_banner_reduction)
-        val description = context.getString(
-            if (isCookieBannerHandlingEnabled) {
-                R.string.reduce_cookie_banner_on_for_site
-            } else {
-                R.string.reduce_cookie_banner_off_for_site
-            },
-        )
-        val icon = if (isCookieBannerHandlingEnabled) {
-            R.drawable.ic_cookies_enabled
-        } else {
-            R.drawable.ic_cookies_disabled
+        val description = when (cookieBannerMode) {
+            CookieBannerMode.COOKIE_BANNER_HANDLING_ENABLE -> context.getString(
+                R.string.reduce_cookie_banner_on_for_site,
+            )
+            CookieBannerMode.COOKIE_BANNER_HANDLING_DISABLE -> context.getString(
+                R.string.reduce_cookie_banner_off_for_site,
+            )
+            CookieBannerMode.REQUEST_UNSUPPORTED_SITE_SUBMITTED -> context.getString(
+                R.string.reduce_cookie_banner_unsupported_site_request_submitted,
+            )
+            CookieBannerMode.SITE_NOT_SUPPORTED -> context.getString(
+                R.string.reduce_cookie_banner_unsupported_site,
+            )
+            CookieBannerMode.COOKIE_BANNER_HANDLING_HIDE -> ""
```

Comment:
We could add a property directly to `CookieBannerMode` called `description` and store the string this will eliminate the when :)

For example: 
```kotlin
enum class CookieBannerMode(@StringRes val description:Int, @DrawableRes val icon: Int) : Parcelable {
    COOKIE_BANNER_HANDLING_ENABLE(R.string.reduce_cookie_banner_on_for_site,R.drawable.ic_cookies_enabled),
    COOKIE_BANNER_HANDLING_DISABLE(R.string.reduce_cookie_banner_off_for_site, R.drawable.ic_cookies_disabled),
    SITE_NOT_SUPPORTED(R.string.reduce_cookie_banner_unsupported_site_request_submitted, R.drawable.ic_cookies_enabled),
    REQUEST_UNSUPPORTED_SITE_SUBMITTED(R.string.reduce_cookie_banner_unsupported_site,R.drawable.ic_cookies_enabled),
}
```

Reply:
For example https://github.com/mozilla-mobile/firefox-android/blob/a488c06ae9ff9d060517da08e2a2db94465dc871/android-components/components/browser/errorpages/src/main/java/mozilla/components/browser/errorpages/ErrorPages.kt#L102